configs/builder_type_cfg.yaml

```diff
@@ -12,36 +12,52 @@
 
 
 builders:
-  - name: npm
-    info: "installing module dependencies & remove dev dependencies"
-    path: "path to config file which override the following default commands"
-    type:
-      - command: npm install
-      - command: npm prune --production
-
-  - name: grunt
-    info: "execute grunt"
-    path: "path to config file which override the following default commands"
-    type:
-      - command: npm install
-      - command: grunt
-
-  - name: golang
-    info: "build golang application"
-    path: "path to config file which override the following default commands"
-    type:
-      - command: go build *.go
-
-  # ---dummy build-before commands testing purpose
-  - name: cds_before
-    info: "test"
-    path: "test"
-    type:
-      - command: echo "-----------test before all modules build----------------"
-
-  # ---dummy commands for build-after testing purpose
-  - name: cds_after
-    info: "test"
-    path: "test"
-    type:
-      - command: echo "-----------test after all modules build----------------"
\ No newline at end of file
+- name: npm
+  info: "installing module dependencies & remove dev dependencies"
+  path: "path to config file which override the following default commands"
+  commands:
+  - command: npm install
+  - command: npm prune --production
+
+- name: grunt
+  info: "execute grunt"
+  path: "path to config file which override the following default commands"
+  commands:
+  - command: npm install
+  - command: grunt
+
+- name: golang
+  info: "build golang application"
+  path: "path to config file which override the following default commands"
+  commands:
+  - command: go build *.go
+
+# ---dummy build-before commands testing purpose
+- name: cds_before
+  info: "test"
+  path: "test"
+  commands:
+  - command: echo "-----------test before all modules build----------------"
+
+# ---dummy commands for build-after testing purpose
+- name: cds_after
+  info: "test"
+  path: "test"
+  commands:
+  - command: echo "-----------test after all modules build----------------"
+
+- name: fetcher
+  info: "packaging already Maven artifacts into MTAR as the modules build results"
+  path: "path to config file which override the following default commands"
+  builder-types:
+  - name: maven
+    commands:
+    - command: mvn dependency:copy -Dartifact={{repo-coordinates}} -DoutputDirectory=./
+
+- name: zip
+  info: "the zip builder archives the source folder into a .zip file and saves it as .mtar without making any changes"
+  path: "the zip builder archives the source folder into a .zip file and saves it as .mtar without making any changes"
+  commands:
+  - command: mbt module zip -m={{module-name}} -s={{source}}
```

Comment:
all other flags to be provided as well, We have to care of some naming convention for "system" properties